features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/ViewEndedMetricDispatcher.kt

```diff
@@ -0,0 +1,164 @@
+    private val samplingRate: Float = DEFAULT_SAMPLE_RATE
+) : ViewMetricDispatcher {
+
+    private var duration: Long? = null
+    private var loadingTime: Long? = null
+    private var metricSent: Boolean = false
+
+    override fun sendViewEnded(invState: ViewInitializationMetricsState, tnsState: ViewInitializationMetricsState) {
+        if (metricSent) {
+            internalLogger.log(
+                InternalLogger.Level.WARN,
+                target = Target.TELEMETRY,
+                messageBuilder = { "Trying to send 'view ended' more than once" }
+            )
+            return
+        }
+
+        internalLogger.logMetric(
+            messageBuilder = { VIEW_ENDED_MESSAGE },
+            additionalProperties = buildMetricAttributesMap(invState, tnsState),
+            samplingRate = samplingRate
+        )
+
+        metricSent = true
+    }
+
+    override fun onDurationResolved(newDuration: Long) {
+        duration = newDuration
+    }
+
+    override fun onViewLoadingTimeResolved(newLoadingTime: Long) {
+        loadingTime = newLoadingTime
+    }
+
+    private fun buildMetricAttributesMap(
+        invState: ViewInitializationMetricsState,
+        tnsState: ViewInitializationMetricsState
+    ): Map<String, Any?> = buildMap {
+        put(KEY_METRIC_TYPE, VALUE_METRIC_TYPE)
+        put(KEY_RVE, buildAttributesMap(invState, tnsState))
+    }
+
+    private fun buildAttributesMap(
+        invState: ViewInitializationMetricsState,
+        tnsState: ViewInitializationMetricsState
+    ): Map<String, Any> = buildMap {
+        putNonNull(KEY_DURATION, duration)
+        if (loadingTime != null) {
+            put(KEY_LOADING_TIME, buildMap { put(KEY_VALUE, loadingTime) })
+        }
+        put(KEY_VIEW_TYPE, toAttributeValue(viewType))
+        put(
+            KEY_TIME_TO_NETWORK_SETTLED,
+            buildMap {
+                put(KEY_VALUE, tnsState.initializationTime)
+                put(KEY_CONFIG, toAttributeValue(tnsState.config))
+                if (tnsState.initializationTime == null) {
+                    putNonNull(KEY_NO_VALUE_REASON, toAttributeValue(tnsState.noValueReason))
+                }
+            }
+        )
+        put(
+            KEY_INTERACTION_TO_NEXT_VIEW,
+            buildMap {
+                put(KEY_VALUE, invState.initializationTime)
+                put(KEY_CONFIG, toAttributeValue(invState.config))
+                if (invState.initializationTime == null) {
+                    putNonNull(KEY_NO_VALUE_REASON, toAttributeValue(invState.noValueReason))
+                }
+            }
+        )
+    }
+
+    companion object {
+        const val DEFAULT_SAMPLE_RATE: Float = 75f
+
+        internal const val VIEW_ENDED_MESSAGE = "[Mobile Metric] RUM View Ended"
+
+        internal const val KEY_METRIC_TYPE = "metric_type"
+
+        private const val VALUE_METRIC_TYPE = "rum view ended"
+
+        internal const val KEY_RVE = "rve"
```

Comment:
maybe it is worth adding a comment saying what "rve" means? Or use full form in the constant name for the clarity, like we do below for `KEY_TIME_TO_NETWORK_SETTLED = "tns"`.